Automattic/newspack-plugin
# fix(woocommerce): update how order meta are updated

### All Submissions:






### Changes proposed in this Pull Request:

[As of WooCommerce 8.2](https://woocommerce.com/document/high-performance-order-storage/), order meta are stored in a separate table and WP's post meta functions are no longer supported.

This PR updates to use methods from the `WC_Order` class.

### How to test the changes in this Pull Request:

1. Enable HPOS
2. Make a donation using the Donate block, on a page with a `?utm_medium=some-medium` param
3. Note the order ID and run `wp wc shop_order get <order-id> --field=meta_data`*
4. Observe the UTM param is listed in `utm` meta key 

\* In my local dev env, I had to specify a user, and I've used `jq` for readability, so in all it was: `wp wc shop_order get <order-id> --user=1 --field=meta_data | jq '.'`

### Other information:
